fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1589,6 +1605,168 @@ private static QueryParameter parseChainedParameter(Class<?> resourceType, Strin
+                }
+
+                // Create new QueryParameter
+                QueryParameter parameter = new QueryParameter(Type.REFERENCE, referenceSearchParameterName, Modifier.TYPE, referencedByResourceTypeName, false, true);
+                if (rootParameter == null) {
+                    rootParameter = parameter;
+                } else {
+                    if (rootParameter.getChain().isEmpty()) {
+                        rootParameter.setNextParameter(parameter);
+                    } else {
+                        rootParameter.getChain().getLast().setNextParameter(parameter);
+                    }
+                }
+
+                if (currentIndex == lastIndex) {
+                    // Add last search parameter
+                    String parameterName = subcomponents.get(2);
+                    if (isChainedParameter(parameterName)) {
+                        QueryParameter lastParameter = parseChainedParameter(referencedByResourceType, parameterName, valuesString);
+                        if (rootParameter.getChain().isEmpty()) {
+                            rootParameter.setNextParameter(lastParameter);
+                        } else {
+                            rootParameter.getChain().getLast().setNextParameter(lastParameter);
+                        }
+                    } else {
+                        String modifierName = null;
+                        Modifier modifier = null;
+                        String modifierResourceTypeName = null;
+
+                        // Check if modifier is specified
+                        int index = parameterName.indexOf(":");
+                        if (index != -1) {
+                            modifierName = parameterName.substring(index + 1);
+                            parameterName = parameterName.substring(0, index);
+                        }
+
+                        SearchParameter searchParameter = getSearchParameter(referencedByResourceType, parameterName);
+                        if (searchParameter == null) {
+                            throw SearchExceptionUtil.buildNewInvalidSearchException(String.format(SEARCH_PARAMETER_NOT_FOUND, parameterName, referencedByResourceTypeName));
+                        }
+                        Type type = Type.fromValue(searchParameter.getType().getValue());
+
+                        if (modifierName != null) {
+                            if (ModelSupport.isResourceType(modifierName)) {
+                                modifier = Modifier.TYPE;
+                                modifierResourceTypeName = modifierName;
+                            } else {
+                                try {
+                                    modifier = Modifier.fromValue(modifierName);
+                                } catch (IllegalArgumentException e) {
+                                    String msg = "Undefined Modifier: " + modifierName;
```

Comment:
we should probably surround the modifierName with single quotes
We might want to URL encode it as well

Reply:
done